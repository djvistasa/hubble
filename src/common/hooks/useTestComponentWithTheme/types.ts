type IUseTestComponentWithTheme = {};

export type { IUseTestComponentWithTheme };
