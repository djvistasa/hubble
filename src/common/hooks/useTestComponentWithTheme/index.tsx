import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../theme";

import { RenderResult } from "@testing-library/react";

import { ReactElement } from "react";
import { QueryClient, QueryClientProvider } from "react-query";

function useTestComponentWithTheme() {
  const renderWithTheme = (component: ReactElement<any, any>): RenderResult => {
    const queryClient = new QueryClient();
    return render(
      <QueryClientProvider client={queryClient}>
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
      </QueryClientProvider>
    );
  };

  return renderWithTheme;
}

export default useTestComponentWithTheme;
