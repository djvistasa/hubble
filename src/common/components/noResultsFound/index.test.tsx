import useTestComponentWithTheme from "@hooks/useTestComponentWithTheme";
import { describe, expect, it } from "vitest";

import NoResultsFound from ".";

describe("NoResultsFound", () => {
  const mountWithTheme = useTestComponentWithTheme();

  it("should render correctly", () => {
    const component = mountWithTheme(<NoResultsFound />);
    expect(component).toMatchSnapshot();
  });
});
