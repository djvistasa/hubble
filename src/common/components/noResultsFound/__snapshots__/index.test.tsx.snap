// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NoResultsFound > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styles__StyledNoResultsFound-sc-wm1ex6-0 jiIOdb"
      >
        <div>
          <svg
            enable-background="new 0 0 32 32"
            height="100"
            id="_x3C_Layer_x3E_"
            viewBox="0 0 32 32"
            width="100"
            xml:space="preserve"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <g
              id="search_x2C__magnifier_x2C__magnifying_x2C__emoji_x2C__happy"
            >
              <g
                id="XMLID_1949_"
              >
                <g
                  id="XMLID_2132_"
                >
                  <path
                    d="M12,14.521h2c0,0.55-0.45,1-1,1S12,15.07,12,14.521z"
                    fill="#263238"
                    id="XMLID_2137_"
                  />
                  <path
                    d="M17.5,13c0.27,0,0.5,0.23,0.5,0.5S17.77,14,17.5,14S17,13.77,17,13.5S17.23,13,17.5,13z     "
                    fill="#263238"
                    id="XMLID_2134_"
                  />
                  <path
                    d="M8.5,13C8.77,13,9,13.23,9,13.5S8.77,14,8.5,14S8,13.77,8,13.5S8.23,13,8.5,13z"
                    fill="#263238"
                    id="XMLID_2133_"
                  />
                </g>
              </g>
              <g
                id="XMLID_1838_"
              >
                <g
                  id="XMLID_4088_"
                >
                  <line
                    fill="none"
                    id="XMLID_4094_"
                    stroke="#455A64"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                    x1="23.43"
                    x2="21.214"
                    y1="23.401"
                    y2="21.186"
                  />
                  <path
                    d="     M29.914,27.086l-3.5-3.5c-0.756-0.756-2.072-0.756-2.828,0C23.208,23.964,23,24.466,23,25s0.208,1.036,0.586,1.414l3.5,3.5     c0.378,0.378,0.88,0.586,1.414,0.586s1.036-0.208,1.414-0.586S30.5,29.034,30.5,28.5S30.292,27.464,29.914,27.086z"
                    fill="none"
                    id="XMLID_4093_"
                    stroke="#455A64"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                  />
                  <circle
                    cx="13"
                    cy="13"
                    fill="none"
                    id="XMLID_4092_"
                    r="11.5"
                    stroke="#455A64"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                  />
                  <path
                    d="     M12,14.521h2c0,0.55-0.45,1-1,1S12,15.07,12,14.521z"
                    fill="none"
                    id="XMLID_4091_"
                    stroke="#455A64"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                  />
                  <path
                    d="     M17.5,13c0.27,0,0.5,0.23,0.5,0.5S17.77,14,17.5,14S17,13.77,17,13.5S17.23,13,17.5,13z"
                    fill="none"
                    id="XMLID_4090_"
                    stroke="#455A64"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                  />
                  <path
                    d="     M8.5,13C8.77,13,9,13.23,9,13.5S8.77,14,8.5,14S8,13.77,8,13.5S8.23,13,8.5,13z"
                    fill="none"
                    id="XMLID_4089_"
                    stroke="#455A64"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                  />
                </g>
                <g
                  id="XMLID_3004_"
                >
                  <line
                    fill="none"
                    id="XMLID_4087_"
                    stroke="#263238"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                    x1="23.43"
                    x2="21.214"
                    y1="23.401"
                    y2="21.186"
                  />
                  <path
                    d="     M29.914,27.086l-3.5-3.5c-0.756-0.756-2.072-0.756-2.828,0C23.208,23.964,23,24.466,23,25s0.208,1.036,0.586,1.414l3.5,3.5     c0.378,0.378,0.88,0.586,1.414,0.586s1.036-0.208,1.414-0.586S30.5,29.034,30.5,28.5S30.292,27.464,29.914,27.086z"
                    fill="none"
                    id="XMLID_3009_"
                    stroke="#263238"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                  />
                  <circle
                    cx="13"
                    cy="13"
                    fill="none"
                    id="XMLID_3008_"
                    r="11.5"
                    stroke="#263238"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                  />
                  <path
                    d="     M12,14.521h2c0,0.55-0.45,1-1,1S12,15.07,12,14.521z"
                    fill="none"
                    id="XMLID_3007_"
                    stroke="#263238"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                  />
                  <path
                    d="     M17.5,13c0.27,0,0.5,0.23,0.5,0.5S17.77,14,17.5,14S17,13.77,17,13.5S17.23,13,17.5,13z"
                    fill="none"
                    id="XMLID_3006_"
                    stroke="#263238"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                  />
                  <path
                    d="     M8.5,13C8.77,13,9,13.23,9,13.5S8.77,14,8.5,14S8,13.77,8,13.5S8.23,13,8.5,13z"
                    fill="none"
                    id="XMLID_3005_"
                    stroke="#263238"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                  />
                </g>
              </g>
            </g>
          </svg>
          <h1>
            No results found
          </h1>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styles__StyledNoResultsFound-sc-wm1ex6-0 jiIOdb"
    >
      <div>
        <svg
          enable-background="new 0 0 32 32"
          height="100"
          id="_x3C_Layer_x3E_"
          viewBox="0 0 32 32"
          width="100"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g
            id="search_x2C__magnifier_x2C__magnifying_x2C__emoji_x2C__happy"
          >
            <g
              id="XMLID_1949_"
            >
              <g
                id="XMLID_2132_"
              >
                <path
                  d="M12,14.521h2c0,0.55-0.45,1-1,1S12,15.07,12,14.521z"
                  fill="#263238"
                  id="XMLID_2137_"
                />
                <path
                  d="M17.5,13c0.27,0,0.5,0.23,0.5,0.5S17.77,14,17.5,14S17,13.77,17,13.5S17.23,13,17.5,13z     "
                  fill="#263238"
                  id="XMLID_2134_"
                />
                <path
                  d="M8.5,13C8.77,13,9,13.23,9,13.5S8.77,14,8.5,14S8,13.77,8,13.5S8.23,13,8.5,13z"
                  fill="#263238"
                  id="XMLID_2133_"
                />
              </g>
            </g>
            <g
              id="XMLID_1838_"
            >
              <g
                id="XMLID_4088_"
              >
                <line
                  fill="none"
                  id="XMLID_4094_"
                  stroke="#455A64"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-miterlimit="10"
                  x1="23.43"
                  x2="21.214"
                  y1="23.401"
                  y2="21.186"
                />
                <path
                  d="     M29.914,27.086l-3.5-3.5c-0.756-0.756-2.072-0.756-2.828,0C23.208,23.964,23,24.466,23,25s0.208,1.036,0.586,1.414l3.5,3.5     c0.378,0.378,0.88,0.586,1.414,0.586s1.036-0.208,1.414-0.586S30.5,29.034,30.5,28.5S30.292,27.464,29.914,27.086z"
                  fill="none"
                  id="XMLID_4093_"
                  stroke="#455A64"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-miterlimit="10"
                />
                <circle
                  cx="13"
                  cy="13"
                  fill="none"
                  id="XMLID_4092_"
                  r="11.5"
                  stroke="#455A64"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-miterlimit="10"
                />
                <path
                  d="     M12,14.521h2c0,0.55-0.45,1-1,1S12,15.07,12,14.521z"
                  fill="none"
                  id="XMLID_4091_"
                  stroke="#455A64"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-miterlimit="10"
                />
                <path
                  d="     M17.5,13c0.27,0,0.5,0.23,0.5,0.5S17.77,14,17.5,14S17,13.77,17,13.5S17.23,13,17.5,13z"
                  fill="none"
                  id="XMLID_4090_"
                  stroke="#455A64"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-miterlimit="10"
                />
                <path
                  d="     M8.5,13C8.77,13,9,13.23,9,13.5S8.77,14,8.5,14S8,13.77,8,13.5S8.23,13,8.5,13z"
                  fill="none"
                  id="XMLID_4089_"
                  stroke="#455A64"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-miterlimit="10"
                />
              </g>
              <g
                id="XMLID_3004_"
              >
                <line
                  fill="none"
                  id="XMLID_4087_"
                  stroke="#263238"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-miterlimit="10"
                  x1="23.43"
                  x2="21.214"
                  y1="23.401"
                  y2="21.186"
                />
                <path
                  d="     M29.914,27.086l-3.5-3.5c-0.756-0.756-2.072-0.756-2.828,0C23.208,23.964,23,24.466,23,25s0.208,1.036,0.586,1.414l3.5,3.5     c0.378,0.378,0.88,0.586,1.414,0.586s1.036-0.208,1.414-0.586S30.5,29.034,30.5,28.5S30.292,27.464,29.914,27.086z"
                  fill="none"
                  id="XMLID_3009_"
                  stroke="#263238"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-miterlimit="10"
                />
                <circle
                  cx="13"
                  cy="13"
                  fill="none"
                  id="XMLID_3008_"
                  r="11.5"
                  stroke="#263238"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-miterlimit="10"
                />
                <path
                  d="     M12,14.521h2c0,0.55-0.45,1-1,1S12,15.07,12,14.521z"
                  fill="none"
                  id="XMLID_3007_"
                  stroke="#263238"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-miterlimit="10"
                />
                <path
                  d="     M17.5,13c0.27,0,0.5,0.23,0.5,0.5S17.77,14,17.5,14S17,13.77,17,13.5S17.23,13,17.5,13z"
                  fill="none"
                  id="XMLID_3006_"
                  stroke="#263238"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-miterlimit="10"
                />
                <path
                  d="     M8.5,13C8.77,13,9,13.23,9,13.5S8.77,14,8.5,14S8,13.77,8,13.5S8.23,13,8.5,13z"
                  fill="none"
                  id="XMLID_3005_"
                  stroke="#263238"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-miterlimit="10"
                />
              </g>
            </g>
          </g>
        </svg>
        <h1>
          No results found
        </h1>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
